pkgs/by-name/pe/peergos/package.nix

```diff
@@ -0,0 +1,39 @@
+{ lib
+, stdenv
+, fetchurl
+, jre
+, makeWrapper
+}:
+
+let
+  version = "0.14.1";
+  peergos = fetchurl {
+    url = "https://github.com/Peergos/web-ui/releases/download/v${version}/Peergos.jar";
+    hash = "sha256-oCsUuFxTAL0vAabGggGhZHaF40A5TLfkT15HYPiKHlU=";
+  };
+in stdenv.mkDerivation rec {
+  inherit version;
+  pname = "peergos";
+  meta = with lib; {
+    description = "A p2p, secure file storage, social network and application protocol";
+    homepage = "https://peergos.org/";
+    # peergos have agpt3 license, peergos-web-ui have gpl3, both are used
+    license = [licenses.agpl3Only licenses.gpl3Only];
+    platforms = platforms.all;
+    maintainers = with maintainers; [ raspher ];
+  };
+
+  dontUnpack = 1;
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  installPhase = ''
+    install -D ${peergos} $out/share/java/peergos.jar
+    makeWrapper ${lib.getExe jre} $out/bin/peergos \
+      --chdir $out/share/peergos/ \
+      --add-flags "-jar $out/share/java/peergos.jar"
+    mkdir -p $out/share/peergos/
```

Comment:
This doesn't make sense to me. What would `cd /nix/store/..-peergos/share/peergos` help with if that directory isn't writable?

Reply:
It won't create `native-libs` folder, please look at mentioned issues

Reply:
I did. So if the directory is not writable then `native-libs` won't be created, which is what we want?
In that case why not `--chdir $out/nowhere`?
I still don't think we should use `--chdir` because then relative paths passed on the command line won't be interpreted correctly.

Reply:
What do you think I should do here? Wait for a fix for that? We can of course

Reply:
The upstream behaviour is to create native-libs in cwd, so we should just go with that behavior until they fix it. Or is the problem that native-libs doesn't work on NixOS? Btw is native-libs downloaded from the internet or compiled on the machine?

Reply:
looks like they have the `.so` file on repo and it's included into the .jar file - their [build.xml](https://github.com/Peergos/Peergos/blob/12ec589e841fd51233bff15d4d3eb27761efca15/build.xml#L81). It seems to be same file loaded into jar on every architecture and unpacked to current directory to use it. Altrough i'm not java expert and may be wrong. You're right, i'll remove the chdir and let unpack this file where they want.